docs/lifecycle/hooks/hooks.md

```diff
@@ -0,0 +1,347 @@
+# Lifecycle Hooks
+
+As many other javascript frontend frameworks, Vue runs through a lifecycle flow where we can identify special or key moments.
+
+<img :src="$withBase('/lifecycle.png')" style="text-align:center"/>
+
+These moments into the lifecycle flow are direclty tied to a very special set of methods, and those methods are known as lifecycle hooks.
+
+Lifecycle hooks allow us to leverage our application's behavior by performing actions such as fetching data, communicating events and so on.
+
+Let's take a look at some useful concepts when it comes to lifecycle.
+
+## Reactivity
+
+Reactivity is probably one of the most important features when working with Vue.js and even with modern javascript frontend frameworks. To put it on simple words, reactivity is a feature that will allow us to keep track of changes in our components/properties into our Vue instance. Eventually, Vue will internally manage and keep track of change notifications and/or dependency changes.
+
+In order to enable this reactivity feature, Vue takes advantage of the `Object.defineProperty` method introduced in ES5, using it to convert every property into a getter/setter.
+
+Below, is a very ilustrative diagram on how Vue handles reactivity:
+
+<img :src="$withBase('/reactivity.png')" style="text-align:center"/>
+
+## Data rendering
+
+On the early steps of this bootcamp, we learnt that the `template` is the suggested place to render our data (or the .html file when it comes to splitted components). However, there are specific cases where we want to use a more in depth approach using the `render` function. Let's take a look at this practical use case comming from the official documentation:
+
+```html
+<template>
+  <h1 v-if="level === 1">
+    <slot></slot>
+  </h1>
+  <h2 v-else-if="level === 2">
+    <slot></slot>
+  </h2>
+  <h3 v-else-if="level === 3">
+    <slot></slot>
+  </h3>
+  <h4 v-else-if="level === 4">
+    <slot></slot>
+  </h4>
+  <h5 v-else-if="level === 5">
+    <slot></slot>
+  </h5>
+  <h6 v-else-if="level === 6">
+    <slot></slot>
+  </h6>
+</template>
+```
+
+```javascript
+module.exports = {
```

Comment:
`module.exports` is how node.js handles module exporting.
In Vue.js as we're relying in webpack we should use

`export default {
...
}`

Reply:
jaja funny thing, when you're a backend and not frontend developer :D